code_samples/recent_activity/src/Command/MonitorRecentContentCreationCommand.php

```diff
@@ -0,0 +1,89 @@
+
+class MonitorRecentContentCreationCommand extends Command
+{
+    protected static $defaultName = 'app:monitor-content-creation';
+
+    private ActivityLogServiceInterface $activityLogService;
+
+    private PermissionResolver $permissionResolver;
+
+    private UserService $userService;
+
+    public function __construct(ActivityLogServiceInterface $activityLogService, PermissionResolver $permissionResolver, UserService $userService)
+    {
+        parent::__construct(self::$defaultName);
+        $this->permissionResolver = $permissionResolver;
+        $this->userService = $userService;
+        $this->activityLogService = $activityLogService;
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription('List last 10 log entry groups with creations in the last hour')
+            ->addOption('filter', 'f', InputOption::VALUE_NONE, 'If only creation log entries should be shown in log groups');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $query = new Query([
+            new Criterion\ObjectCriterion(Content::class),
+            new Criterion\ActionCriterion([ActivityLogServiceInterface::ACTION_CREATE]),
+            new Criterion\LoggedAtCriterion(new \DateTime('- 1 hour'), Criterion\LoggedAtCriterion::GTE),
+        ], [new LoggedAtSortClause(LoggedAtSortClause::DESC)], 0, 10);
+
+        $filterLogs = $input->getOption('filter');
+
+        $io = new SymfonyStyle($input, $output);
+
+        $this->permissionResolver->setCurrentUserReference($this->userService->loadUserByLogin('admin'));
+
+        foreach ($this->activityLogService->findGroups($query) as $activityLogGroup) {
+            if ($activityLogGroup->getSource()) {
+                $io->section($activityLogGroup->getSource()->getName());
+            }
+            if ($activityLogGroup->getDescription()) {
+                $io->text($activityLogGroup->getDescription());
+            }
+            $table = [];
+            foreach ($activityLogGroup->getActivityLogs() as $activityLog) {
+                if (!$filterLogs || $activityLog->getAction() === ActivityLogServiceInterface::ACTION_CREATE) {
+                    /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                    $content = $activityLog->getRelatedObject();
+                    $name = $content && $content->getName() && $content->getName() !== $activityLog->getObjectName() ? "“{$content->getName()}” (formerly “{$activityLog->getObjectName()}”)" : "“{$activityLog->getObjectName()}”";
+                    $table[] = [
+                        $activityLogGroup->getLoggedAt()->format(\DateTime::ATOM),
+                        $activityLog->getObjectId(),
+                        $name,
+                        $activityLog->getAction(),
+                        $activityLogGroup->getUser()->login,
+                    ];
+                }
```

Comment:
Prefer early "return" ("continue" in this case, but the same rule).

Also, what is the purpose of `$filterLogs` variable? It doesn't seem to be used.

```suggestion
                if ($filterLogs && $activityLog->getAction() !== ActivityLogServiceInterface::ACTION_CREATE) {
                    continue;
                }

                /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
                $content = $activityLog->getRelatedObject();
                $name = $content && $content->getName() && $content->getName() !== $activityLog->getObjectName() ? "“{$content->getName()}” (formerly “{$activityLog->getObjectName()}”)" : "“{$activityLog->getObjectName()}”";
                $table[] = [
                    $activityLogGroup->getLoggedAt()->format(\DateTime::ATOM),
                    $activityLog->getObjectId(),
                    $name,
                    $activityLog->getAction(),
                    $activityLogGroup->getUser()->login,
                ];
```

Reply:
`$filterLogs = $input->getOption('filter');`

I could go with a `php bin/console app:monitor-content-creation --filter` example of output to show that there is less. But it will be too heavy.

I should remove this. Especially since there will be something to achieve this in the future.

Reply:
`--filter` && `$filterLogs` removed as incoming `ActivityLogService::findLogs` will do it better in next version.